### PR TITLE
Override RC file with CLI params

### LINKED ISSUE #12: override rc file with params
add the ability to override values in the `.topcoderrc` file with params at run time. ie:
`tc submit --username lazybaer --challengeIds=123456789,987654321`

### PR DESCRIPTION
**Pull request by sharathkumaranbu**
Fixes #12

### CODE REVIEW

File: .topcoderrc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (ThomasKranitsas) [inline .topcoderrc]: @sharathkumaranbu do we need that file here?

Comment (sharathkumaranbu) [reply .topcoderrc]: I thought could be for an example if someone looks at the source code, we have it in documentation any way. We can remove it if required

File: src/common/helper.js [hunk 2]
```diff
@@ -20,12 +20,15 @@ const schemaForRC = Joi.object({
...
  * @param {String} filename the name of the rc file
+ * @param {Object} cliParams CLI params passed to the program
  * @returns {Object} the rc object
  */
-function readFromRCFile (filename) {
+function readFromRCFile (filename, cliParams) {
   logger.info('Reading from topcoder rc file...')
   const rcObject = JSON.parse(fs.readFileSync(filename).toString())
-  return validateRCObject(rcObject)
+  cliParams.challengeIds = cliParams.challengeIds.split(',')
```
Comment (ThomasKranitsas) [inline src/common/helper.js]: This is complaining if you don't pass the challenge IDs as an argument param.

```error: Cannot read property 'split' of undefined```

File: src/services/uploadSubmissionService.js [hunk 3]
```diff
@@ -10,15 +10,19 @@ const logger = require('../common/logger')
...
- *
+ * @param {String} currDir Path to current directory
+ * @param {Object} cliParams CLI params passed to the program
  * @returns {Array} Uploaded submissions
  */
-async function smart (prefix) {
-  const { username, password, challengeIds } = helper.readFromRCFile(path.join(prefix, constants.rc.name))
+async function smart (currDir, cliParams) {
+  const { username, password, challengeIds } = helper.readFromRCFile(path.join(currDir, constants.rc.name), cliParams)
+  console.log(username)
```
Comment (ThomasKranitsas) [inline src/services/uploadSubmissionService.js]: Can you please remove the console.log()

Comment (sharathkumaranbu) [reply src/services/uploadSubmissionService.js]: Done in latest commit

File: src/common/helper.js (hunk 2 above)
Comment (sharathkumaranbu) [reply src/common/helper.js]: Fixed

**Pull request merged**